nuxeo-features/nuxeo-automation/nuxeo-automation-core/src/test/java/org/nuxeo/ecm/automation/core/operations/document/MoveToColdStorageTest.java

```diff
@@ -0,0 +1,127 @@
+import static org.junit.Assert.fail;
+import static org.nuxeo.ecm.core.DummyThumbnailFactory.DUMMY_THUMBNAIL_CONTENT;
+import static org.nuxeo.ecm.core.blob.ColdStorageHelper.COLD_STORAGE_CONTENT_PROPERTY;
+import static org.nuxeo.ecm.core.blob.ColdStorageHelper.FILE_CONTENT_PROPERTY;
+import static org.nuxeo.ecm.core.schema.FacetNames.COLD_STORAGE;
+
+import java.io.IOException;
+import java.io.Serializable;
+import java.util.Map;
+
+import javax.inject.Inject;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.nuxeo.ecm.automation.AutomationService;
+import org.nuxeo.ecm.automation.OperationContext;
+import org.nuxeo.ecm.automation.OperationException;
+import org.nuxeo.ecm.core.api.Blob;
+import org.nuxeo.ecm.core.api.Blobs;
+import org.nuxeo.ecm.core.api.CoreSession;
+import org.nuxeo.ecm.core.api.DocumentModel;
+import org.nuxeo.ecm.core.api.NuxeoException;
+import org.nuxeo.ecm.core.test.CoreFeature;
+import org.nuxeo.runtime.test.runner.Deploy;
+import org.nuxeo.runtime.test.runner.Features;
+import org.nuxeo.runtime.test.runner.FeaturesRunner;
+
+/**
+ * @since 11.1
+ */
+@RunWith(FeaturesRunner.class)
+@Features(CoreFeature.class)
+@Deploy("org.nuxeo.ecm.automation.core")
+@Deploy("org.nuxeo.ecm.core.test.tests:OSGI-INF/test-coldstorage-contrib.xml")
+public class MoveToColdStorageTest {
+
+    protected static final String FILE_CONTENT = "foo and boo";
+
+    @Inject
+    protected CoreSession session;
+
+    @Inject
+    protected AutomationService automationService;
+
+    @Test
+    public void shouldMoveToColdStorage() throws OperationException, IOException {
+        DocumentModel documentModel = createDocument(true);
+        moveContentToColdStorage(documentModel);
+    }
+
+    @Test
+    public void shouldFailWhenMovingDocumentBlobAlreadyInColdStorage() throws OperationException, IOException {
+        DocumentModel documentModel = createDocument(true);
+        // make a move
+        moveContentToColdStorage(documentModel);
+        try {
+            // try to make a second move
+            moveContentToColdStorage(documentModel);
+            fail("Should fail because the content is already in cold storage");
+        } catch (NuxeoException ne) {
+            assertEquals(SC_CONFLICT, ne.getStatusCode());
+        }
+    }
+
+    @Test
+    public void shouldFailWhenMovingToColdStorageDocumentWithoutContent() throws OperationException, IOException {
+        DocumentModel documentModel = createDocument(false);
+        try {
+            moveContentToColdStorage(documentModel);
+            fail("Should fail because there is no main content associated with the document");
+        } catch (NuxeoException ne) {
+            assertEquals(SC_NOT_FOUND, ne.getStatusCode());
+        }
+    }
+
+    protected void moveContentToColdStorage(DocumentModel documentModel) throws OperationException, IOException {
+        assertFalse(documentModel.hasFacet(COLD_STORAGE));
```

Comment:
I think you should remove this, because `shouldFailWhenMovingToColdStorageDocumentWithoutContent` will fail not for the expected reason.